wrappers/golang/curves/bls12377/vec_ops_test.go

```diff
@@ -31,3 +32,37 @@ func TestVecOps(t *testing.T) {
 
 	assert.Equal(t, a, out3)
 }
+
+func TestTranspose(t *testing.T) {
+
+	rowSize := 1 << 6
+	columnSize := 1 << 8
+	onDevice := false
+	isAsync := true
+
+	matrix := GenerateScalars(rowSize * columnSize)
+
+	out := make(core.HostSlice[ScalarField], rowSize*columnSize)
+	out2 := make(core.HostSlice[ScalarField], rowSize*columnSize)
+
+	ctx, _ := cr.GetDefaultDeviceContext()
+
+	TransposeMatrix(matrix, out, columnSize, rowSize, ctx, onDevice, isAsync)
+	TransposeMatrix(out, out2, rowSize, columnSize, ctx, onDevice, isAsync)
```

Comment:
Since `isAsync` is `true`,  a `syncStream` should be called before performing anything else after the async calls to ensure that the operations have finished

Reply:
Yes my bad I only set isAsync to true to see if it would break stuff